dbc2val/mapping.md

```diff
@@ -0,0 +1,308 @@
+
+The idea of overlays is to specify additions in a separate file and then apply this on top of
+a VSS tree. An example dbc overlay exists in [dbc_overlay.vspec](dbc_overlay.vspec).
+An overlay file is a VSS `*.vspec` file, which shall in itself be a valid VSS tree.
+The VSS tooling requires that type and datatype are defined in addition to the DBC specific data
+needed by the DBC feeder, like in the example below: 
+
+```yaml
+Vehicle.Speed:
+  type: sensor
+  datatype: float
+  dbc:
+    signal: DI_uiSpeed
+    interval_ms: 5000
+```
+
+To create a VSS JSON file considering the overlay file [vss-tools](https://github.com/COVESA/vss-tools)
+must be used. Two alternatives exist. The first alternative is to use raw `*.vspec` file in the
+[VSS repository](https://github.com/COVESA/vehicle_signal_specification) as base. To use that method
+you must clone the repository, update submodules and then generate VSS JSON like in the example below:
+
+```
+git submodule update --init
+vss-tools/vspec2json.py -e dbc -o dbc_overlay.vspec --no-uuid ./spec/VehicleSignalSpecification.vspec vss_dbc.json
+```
+
+An alternative approach is download a tar archive from an [official VSS release](https://github.com/COVESA/vehicle_signal_specification/releases)
+and use the included Yaml file as base.
+
+```
+vss-tools/vspec2json.py -e dbc -o dbc_overlay.vspec --no-uuid vss_rel_3.0.yaml vss_dbc.json
+```
+
+_Note: The dbc feeder relies on correct VSS information in the JSON file. This means that if KUKSA.val Databroker VSS JSON file updated, then the file used in DBC-feeder possibly needs to be updated as well._
+
+
+## Mapping syntax
+
+The syntax for a DBC definition of a signal in an overlay is specified below.
+The syntax if information instead is added directly to a VSS JSON file is similar, but not described here.
+See [vss_dbc.json](vss_dbc.json) for examples on DBC specification in JSON format.
+Search for `dbc` to find the signals where DBC mapping has been defined.
+If a signal does not have DBC mapping it will be ignored by the DBC feeder.
+
+Syntax
+
+```yaml
+<VSS Signal name>:
+  type: <VSS type>
+  datatype: <VSS datatype>
+  dbc:
+    signal: <DBC signal name>
+    [interval_ms: <interval in milliseconds>]
+    [on_change: {True|False}]
+    [transform: ...]
+```
+
+Specifying DBC signal name is mandatory. It must correspond to a DBC signal name defined in a DBC file.
+By default the DBC feeder use the [Model3CAN.dbc](Model3CAN.dbc) example file.
+
+`interval_ms` and `on_change` are optional and control under which conditions a value shall be forwarded.
+The `interval_ms` value indicates the minimum interval between signals in milliseconds.
+The `on_change: True` argument specifies that the VSS signal only shall be sent if the DBC value has changed.
+If none of them are specified it corresponds to `interval_ms: 1000, on_change: False`.
+If only `on_change: True` is specified it corresponds to `interval_ms: 0, on_change: True`
+
```

Comment:
if on-change is true and an interval is given (not sure that is alway useful), how is the logic? I.e.

I open the door-> set open

Then the signal toggles millsions of times within one second, (interval_ms: 1000), so I assume it is not forwarded?
Then after the 1000ms IF the signal is coming again, is "on-change" based on the last reported or on the last observed value? Observed might be a problem because door->open=reported to VSS, door->closed not bcasue within timeout, and then will never be reported, if compare to last observed state.

Might be a corner case, but at least we should now - and preferably describe - what happens here

Reply:
I try to describe it a bit further down, will update it to make it clearer. The filtering is done in two phases. This partially depends on that we have a queue in between. I assume math-transform can be costly from performance perspective, and that is why we only do it time condition is fulfilled.

On_Change is always compared with last value sent to KUKSA.val. So theoretically, f you are very quick in opening/closing your door and you have an interval of 1000 ms then there is a risk that the opening not will be reported to KUKSA.val as we only "poll/sample" values every 1000 ms. 

* If there is a time condition the time of the observation is compared with the stored value.
  If the time difference is bigger than the explicitly or implictly defined interval
  the stored time for the VSS-DBC combination is updated
  and evaluation continue with the next step.
* The DBC value is then transformed to VSS value. If transformation fails the signal is discarded.
* After transformation, if there is a change condition, the stored value is compared with the
  new value. If they are equal the signal is discarded.